support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -135,9 +131,12 @@ public HttpResponse doSubmit(StaplerRequest request) throws IOException, Servlet
      * REST endpoint to submit the input.
      */
     @RequirePOST
-    public HttpResponse doProceed(StaplerRequest request) throws IOException, ServletException, InterruptedException {
+    public HttpResponse doProceed(StaplerRequest request, User user) throws IOException, ServletException, InterruptedException {
         preSubmissionCheck();
-
+        if (user!=null){
+            run.addAction(new ApproverAction(user.getDisplayName()));
```

Comment:
The `User.displayName` could change at any time, and cannot be used to uniquely identify this person. You should instead store the `User.id`. (That also allows the `summary.jelly` to hyperlink to the `/user/*` page, assuming the user still exists when it is rendered. [Here](https://github.com/jenkinsci/jenkins/blob/419d670cd0f672b0b3eba6ad4bcc97e8c6948424/core/src/main/resources/hudson/model/Cause/UserIdCause/description.jelly) is an example.)